From 6943ba674949088b3db6830e58f748d1780885bf Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Thu, 7 Sep 2017 17:32:59 +0000
Subject: [PATCH] Add LuneOS specific switches

* Disable some sandbox capabilities, which are incompatible with our (old) kernels
* Disable OpenGL ES 3, as Qt's shared context uses OpenGL ES 2
* Enable flash/widevine plugins
---
 src/core/web_engine_context.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/core/web_engine_context.cpp b/src/core/web_engine_context.cpp
index 9d993b7..7f0382f 100644
--- a/src/core/web_engine_context.cpp
+++ b/src/core/web_engine_context.cpp
@@ -324,6 +324,18 @@ WebEngineContext::WebEngineContext()
     // Enabled on OS X and Linux but currently not working. It worked in 5.7 on OS X.
     parsedCommandLine->AppendSwitch(switches::kDisableGpuMemoryBufferVideoFrames);
 
+    //// LuneOS specific
+    // Disable some sandbox capabilities, which are incompatible with our (old) kernels
+    parsedCommandLine->AppendSwitch(switches::kDisableNamespaceSandbox);
+    parsedCommandLine->AppendSwitch(switches::kDisableSeccompFilterSandbox);
+    // Disable OpenGL ES 3, as Qt's shared context uses OpenGL ES 2
+    parsedCommandLine->AppendSwitch(switches::kDisableES3GLContext);
+    // Add switches to enable flash/widevine plugins
+    parsedCommandLine->AppendSwitchASCII("ppapi-flash-path", "/usr/lib/chromium/libpepflashplayer.so");
+    parsedCommandLine->AppendSwitchASCII("ppapi-flash-version", "26.0.0.151");
+    parsedCommandLine->AppendSwitchASCII(switches::kRegisterPepperPlugins, "/usr/lib/chromium/libwidevinecdmadapter.so;application/x-ppapi-widevine-cdm");
+    //// End of LuneOS specific
+
 #if defined(Q_OS_MACOS)
     // Accelerated decoding currently does not work on macOS due to issues with OpenGL Rectangle
     // texture support. See QTBUG-60002.
-- 
2.7.4

